### PR TITLE
Corrections

### PR DESCRIPTION
**Pull request by Quentin01**
Remake of #1

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+var express = require('express');
+var router = express.Router();
+var version;
```
Comment (Quentin01) [inline api/index.js]: Why is `version` declared here?

File: api/v1/customTypes/formats.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        mongoose.SchemaType.call(this, key, options, 'Formats');
+    };
+
+    Formats.prototype = Object.create(mongoose.SchemaType.prototype);
+
+    Formats.prototype.cast = function (val) {
+        let _val = val;
+
+        if (!(_val instanceof Array)) {
+            throw new Error(_val + " is not a array");
```
Comment (Quentin01) [inline api/v1/customTypes/formats.js]: "an array"

File: api/v1/customTypes/int10.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+module.exports = function (mongoose) {
+    function Int10(key, options) {
+        mongoose.SchemaType.call(this, key, options, 'Int10');
```
Comment (Quentin01) [inline api/v1/customTypes/int10.js]: I think it's really better to rename that "uint10" for the future users / developers to understand correctly that we can't have negative values

File: api/v1/customTypes/int12.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+module.exports = function (mongoose) {
+    function Int12(key, options) {
+        mongoose.SchemaType.call(this, key, options, 'Int12');
```
Comment (Quentin01) [inline api/v1/customTypes/int12.js]: I think it's really better to rename that "uint12" for the future users / developers to understand correctly that we can't have negative values

File: api/v1/models/gpu/index.js [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+                "maxImageDimension3D": Number,
+                "maxImageDimensionCube": Number,
+                "maxImageArrayLayers": Number,
+                "maxTexelBufferElements": Number,
+                "maxUniformBufferRange": Number,
+                "maxStorageBufferRange": Number,
+                "maxPushConstantsSize": Number,
+                "maxMemoryAllocationCount": Number,
+                "maxSamplerAllocationCount": Number,
+                "bufferImageGranularity": String,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+                "sparseAddressSpaceSize": String,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+                "maxDrawIndexedIndexValue": Number,
+                "maxDrawIndirectCount": Number,
+                "maxSamplerLodBias": Number,
+                "maxSamplerAnisotropy": Number,
+                "maxViewports": Number,
+                "maxViewportDimensions": [Number],
+                "viewportBoundsRange": [Number],
+                "viewportSubPixelBits": Number,
+                "minMemoryMapAlignment": Number,
+                "minTexelBufferOffsetAlignment": String,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+                "minUniformBufferOffsetAlignment": String,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+                "minStorageBufferOffsetAlignment": String,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+                "maxCullDistances": Number,
+                "maxCombinedClipAndCullDistances": Number,
+                "discreteQueuePriorities": Number,
+                "pointSizeRange": [Number],
+                "lineWidthRange": [Number],
+                "pointSizeGranularity": Number,
+                "lineWidthGranularity": Number,
+                "strictLines": Number,
+                "standardSampleLocations": Number,
+                "optimalBufferCopyOffsetAlignment": String,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+                "optimalBufferCopyRowPitchAlignment": String,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+                "nonCoherentAtomSize": String
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Number`

File: api/v1/models/gpu/index.js [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+                "maxTexelOffset": Number,
+                "minTexelGatherOffset": Number,
+                "maxTexelGatherOffset": Number,
+                "minInterpolationOffset": Number,
+                "maxInterpolationOffset": Number,
+                "subPixelInterpolationOffsetBits": Number,
+                "maxFramebufferWidth": Number,
+                "maxFramebufferHeight": Number,
+                "maxFramebufferLayers": Number,
+                "framebufferColorSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 14]
```diff
@@ -0,0 +1,226 @@
...
+                "framebufferDepthSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+                "framebufferStencilSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+                "framebufferNoAttachmentsSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+                "maxColorAttachments": Number,
+                "sampledImageColorSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 18]
```diff
@@ -0,0 +1,226 @@
...
+                "sampledImageIntegerSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 19]
```diff
@@ -0,0 +1,226 @@
...
+                "sampledImageDepthSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 20]
```diff
@@ -0,0 +1,226 @@
...
+                "sampledImageStencilSampleCounts": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `[String]`

File: api/v1/models/gpu/index.js [hunk 21]
```diff
@@ -0,0 +1,226 @@
...
+                "storageImageSampleCounts": Number,
+                "maxSampleMaskWords": Number,
+                "timestampComputeAndGraphics": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Boolean`

File: api/v1/models/gpu/index.js [hunk 22]
```diff
@@ -0,0 +1,226 @@
...
+                "timestampPeriod": Number,
+                "maxClipDistances": Number,
+                "maxCullDistances": Number,
+                "maxCombinedClipAndCullDistances": Number,
+                "discreteQueuePriorities": Number,
+                "pointSizeRange": [Number],
+                "lineWidthRange": [Number],
+                "pointSizeGranularity": Number,
+                "lineWidthGranularity": Number,
+                "strictLines": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Boolean`

File: api/v1/models/gpu/index.js [hunk 23]
```diff
@@ -0,0 +1,226 @@
...
+                "standardSampleLocations": Number,
```
Comment (Quentin01) [inline api/v1/models/gpu/index.js]: `Boolean`

File: config/index.js [hunk 24]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+    server: {
+        port: process.env.PORT || 5000
+    },
+    db: {
+        uri_mongodb: process.env.URI_MONGODB,
```
Comment (Quentin01) [inline config/index.js]: The convention isn't `MONGODB_URI`?

Comment (Quentin01) [reply config/index.js]: And do you really want to repeat `mongodb` for all the field of `db`?

File: server.js [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+
+var config = require('./config');
+var server = express();
+
+var db = mongoose.connect(config.db.uri_mongodb, (err) => {
+    if (err) {
+        console.log(err);
+        return;
+    }
+	console.log("Is connected to the db");
```
Comment (Quentin01) [inline server.js]: Fail indent

File: api/index.js (hunk 1 above)
Comment (yoannpicquenot) [reply api/index.js]: I forgot to remove it...

File: api/v1/customTypes/formats.js (hunk 2 above)
Comment (yoannpicquenot) [reply api/v1/customTypes/formats.js]: Ah yeah

File: api/v1/models/gpu/index.js [hunk 26]
```diff
@@ -0,0 +1,226 @@
...
+require("../../customTypes/int10.js")(mongoose);
+require("../../customTypes/formats.js")(mongoose);
+
+module.exports = function () {
+    var schema = new mongoose.Schema({
+        "properties": {
+            "apiVersion": {
+                "major": mongoose.Schema.Types.Int10,
+                "minor": mongoose.Schema.Types.Int10,
+                "patch": mongoose.Schema.Types.Int12
```
Comment (yoannpicquenot) [inline api/v1/models/gpu/index.js]: mongoose.Schema.Types.Uint10,
mongoose.Schema.Types.Uint10
mongoose.Schema.Types.Uint12

File: api/v1/models/gpu/index.js [hunk 27]
```diff
@@ -0,0 +1,226 @@
...
+            },
+            "driverVersion": {
+                "major": mongoose.Schema.Types.Int10,
+                "minor": mongoose.Schema.Types.Int10,
+                "patch": mongoose.Schema.Types.Int12
```
Comment (yoannpicquenot) [inline api/v1/models/gpu/index.js]: mongoose.Schema.Types.Uint10,
mongoose.Schema.Types.Uint10
mongoose.Schema.Types.Uint12

**Review by yoannpicquenot — COMMENTED** [review]
Hello !

I think my PR is ready to be merged !

Sincerely,

Yoann Picquenot

**Pull request merged**